BuildOpenCV.cmake

```diff
@@ -16,28 +16,56 @@ endif ()
 if (NOT EXISTS ${OPENCV_DOWNLOAD_DIR}/opencv-${OPENCV_VERSION}/CMakeLists.txt)
     message(STATUS "OpenCV not extracted into ${OPENCV_DOWNLOAD_DIR}")
 
-    if (NOT EXISTS ${OPENCV_DOWNLOAD_DIR}/${OPENCV_VERSION}.zip)
+    if (NOT EXISTS ${OPENCV_DOWNLOAD_DIR}/opencv-${OPENCV_VERSION}.zip)
         message(STATUS "OpenCV not downloaded into ${OPENCV_DOWNLOAD_DIR}")
 
         message(STATUS "Downloading OpenCV")
         file(
                 DOWNLOAD
                 https://github.com/opencv/opencv/archive/refs/tags/${OPENCV_VERSION}.zip
-                ${OPENCV_DOWNLOAD_DIR}/${OPENCV_VERSION}.zip
+                ${OPENCV_DOWNLOAD_DIR}/opencv-${OPENCV_VERSION}.zip
                 SHOW_PROGRESS
         )
         message(STATUS "Downloading OpenCV - done")
+
```

Comment:
The base rule -- in PRs there should not be such useless changes. They usually appear when you do `git add` for the whole file. Read about command `git add -p` and start using it later to avoid such situations. That is classical professional git skills.